pkg/csi/service/vanilla/controller_helper.go

```diff
@@ -314,8 +314,9 @@ func getCandidateDSInTopologyForFileVolumes(ctx context.Context,
 		}
 		accessibleDatastoresForHosts, err := vsphere.GetAllAccessibleDatastoresForHosts(ctx, hostMoRefs)
 		if err != nil {
-			return nil, logger.LogNewErrorf(log, "failed to get accessible datastores for hosts: %+v "+
+			log.Errorf("failed to get accessible datastores for hosts: %+v "+
```

Comment:
Any reason why we are making this change? If there is an issue with one of the topology segments, we will always silently fail and not provision in that segment at all

Reply:
@shalini-b  here if we do not get error but accessibleDatastoresForHosts is zero then we move to next topology segment?

Reply:
@shalini-b we cam across an issue where volume provisioning fails completely if we cannot create volume in the given topology segment. We never check other segments even though it is possible to provision volumes on them.

Consider the following deployment:

-region-1
 - zone-1
   - building-1
     - level-1
       - rack-1
       - rack-2
   - building-2
     - level-2
       - rack-3


SC has allowed topology as  this:

```
allowedTopologies:
  - matchLabelExpressions:
      - key: topology.csi.vmware.com/k8s-rack
        values:
          - rack-1
          - rack-3
      - key: topology.csi.vmware.com/k8s-region
        values:
          - region-1
      - key: topology.csi.vmware.com/k8s-zone
        values:
          - zone-1
      - key: topology.csi.vmware.com/k8s-level
        values:
          - level-1
          - level-2
      - key: topology.csi.vmware.com/k8s-building
        values:
          - building-1
          - building-2
```

In this case we will get all combinations of topology segments winclusing region-1, zone-1, building-1, level-2, rack-3 which does not exist.